### PR TITLE
[cpu][inductor] improve cpu vec implementations of log

### LINKED ISSUE #110611: torch.compile CPU backend is slower than eager for several transcendental functions
I tried a handful of somewhat important transcendentals (sigmoid, tanh, log, logit), and they all appear to be slower than eager.  On my 96c/192t Intel(R) Xeon(R) Platinum 8339HC CPU:

```
$ OMP_NUM_THREADS=1 numactl -C 94 python bench.py
log
eager:    363.3282147347927
compiled: 1091.864187270403
sigmoid
eager:    756.7794565111399
compiled: 872.3638635128736
logit
eager:    731.156948953867
compiled: 1594.5508629083633
tanh
eager:    475.9031366556883
compiled: 867.8419776260853

$ python bench.py
log
eager:    28.686294332146645
compiled: 106.62426799535751
sigmoid
eager:    91.70412458479404
compiled: 108.63539576530457
logit
eager:    25.92829428613186
compiled: 88.3821751922369
tanh
eager:    28.41925248503685
compiled: 113.94776217639446
```

```
import time

import torch

def do_bench(fn, iters=1000):
    fn()  # warmup
    s = time.perf_counter()
    for _ in range(iters):
        fn()
    e = time.perf_counter()

    return (e - s) /  iters * 1e6

x = torch.clamp(torch.rand(1 << 20), 1e-6, 1 - 1e-6)  # good range for log, logit

for fn in [torch.log,
           torch.sigmoid,
           torch.logit,
           torch.tanh,
           ]:
    cfn = torch.compile(fn)
    print(fn.__name__)

    eager = do_bench(lambda: fn(x))
    compiled = do_bench(lambda: cfn(x))

    print(f"eager:    {eager}")
    print(f"compiled: {compiled}")
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### PR DESCRIPTION
**Pull request by Valentine233**
Fixes #110611.

The current Torchinductor's `log` implementations will call `sleef` functions in `aten::Vec` which show worse performance than Aten's `log` implementations that invoke `MKL` functions. The reason is that the `sleef` algorithms sacrifice performance in order to have a higher precision. This PR changes Torchinductor's `log` implementations from the `sleef` functions with `1.0` ULP error bound to the ones with `3.5` ULP error bound.

**Performance**
Machine: ICX

The original perf number, perf with `Sleef_logf16_u10`:
```bash
numactl -C0 python test.py
log
eager:    368.8463559374213
compiled: 616.8672097846866
logit
eager:    565.499295014888
compiled: 1010.4096410796046
```

Perf with `Sleef_logf16_u35`:
```bash
numactl -C0 python test.py
log
eager:    364.8629770614207
compiled: 360.2141812443733
logit
eager:    562.3160391114652
compiled: 545.2622110024095
```

**Accuracy**
error_bound | tol=1e-6 | tol=1e-7
-- | -- | --
1.0 ULP | PASS | FAIL
3.5 ULP | PASS | FAIL

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @mlazos @soumith @yanboliang @anijain2305 @Xia-Weiwen @desertfire

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Comment by bertmaher** [comment]
How does this approach compare in accuracy to MKL's log?  The docs for MKL-VML are pretty obtuse but it sounds like the default precision is <1 ulp.  Why is it okay now to use a reduced precision function here?

**Comment by izaitsevfb** [comment]
We decided to revert this change temporarily, as it failed internal test with a numerical mismatch, which is suspicious. We've contacted test owners to clarify.

**Comment by Valentine233** [comment]
> How does this approach compare in accuracy to MKL's log? The docs for MKL-VML are pretty obtuse but it sounds like the default precision is <1 ulp. Why is it okay now to use a reduced precision function here?

For accuracy comparison, I compared the result of inductor (using Sleef's log) with that of eager (using MKL's log) and see if the two results are close. The conclusion is that the accuracy check would fail with `tol=1e-7` and would pass with `tol=1e-6`, whether we use `Sleef_logf16_u10` or `Sleef_logf16_u35`.

**Comment by Valentine233** [comment]
@izaitsevfb @bertmaher Hi, could you pls provide with the internal test which failed with a numerical mismatch? I would like to reproduce it and see how to fix the problem. Thanks!

**Comment by jgong5** [comment]
> We decided to revert this change temporarily, as it failed internal test with a numerical mismatch, which is suspicious. We've contacted test owners to clarify.

@izaitsevfb @bertmaher Is the internal failure confirmed to be caused by this change? If so, how can I repro the problem locally for a fix? Thanks!

**Comment by izaitsevfb** [comment]
> > We decided to revert this change temporarily, as it failed internal test with a numerical mismatch, which is suspicious. We've contacted test owners to clarify.
> 
> @izaitsevfb @bertmaher Is the internal failure confirmed to be caused by this change? If so, how can we repro the problem locally for a fix? Thanks!

Not yet, apologies for the delay.

**Comment by bertmaher** [comment]
Even setting aside the internal test, this seems like a sketchy change, as we're losing 2.5 ulp precision, and I'm not convinced that's ok.  `atol` and `rtol` are sort of crude when looking at few-ulp differences and I would prefer comparisons that use something like `np.testing.assert_array_max_ulp(a, b)` to get a sense of how this compares to MKL precision.

**Comment by marxhxxx** [comment]
Hey. This change breaks one of our internal uts. This ut is important for us to do some model verification. We are verifying the relationship between this change and that ut. It might take some time. We might hold this change until we find the root cause. Sorry for the inconvenience.

**Review by lezcano — COMMENTED** [review]
Note that all the functions in `Vectorized` use `_u10` precision, so this change were to be approved, I think it should be done for all the functions in `Vectorized`.
In all fairness, I'm not sure this is something we can afford or even want to do.

**Comment by jgong5** [comment]
> Note that all the functions in `Vectorized` use `_u10` precision, so this change were to be approved, I think it should be done for all the functions in `Vectorized`.
> In all fairness, I'm not sure this is something we can afford or even want to do.

We changed the bound to u35 for sin/cos previously: https://github.com/pytorch/pytorch/pull/94577. But it doesn't seem correct either since we are using high-accuracy MKL kernels for these math functions which have u10 bound. I guess we have to follow the similar approach like erf optimization (https://github.com/pytorch/pytorch/pull/91613) for these functions: sin/cos/log.

**Comment by Valentine233** [comment]
Checking the accuracy with `np.testing.assert_array_max_ulp` suggested by @bertmaher, we do find the ULP differences:

| Operator | Sleef function | Max ULP |
| -------------- | -------- | ---------- |
| log | sleef10 | 1.0 |
|  | sleef35 | 2.0 |
| logit | sleef10 | 1.0 |
|  | sleef35 | 3.0 |
| cos | sleef10 | 1.0 |
|  | sleef35 | 2.0 |
| sin | sleef10 | 1.0 |
|  | sleef35 | 1.0 |

We decide to re-implement these kernels as @jgong5 said.
